plugins/modules/proxmox_snap.py

```diff
@@ -228,7 +250,9 @@ def snapshot_create(self, vm, vmid, timeout, snapname, description, vmstate, unb
             timeout -= 1
         if vm['type'] == 'lxc' and unbind is True and mountpoints:
             self._container_mp_restore(vm, vmid, timeout, unbind, mountpoints, vmstatus)
-        return False
+
+        self.snapshot_trim(vm, vmid, trim)
```

Comment:
Right now the trimming is only done when a snapshot is really created. If the snapshot to be created already exists, no trimming is done.

Reply:
True. This is like I intended it. However, I'm open for discussions. In theory, If we do NOT create a new snapshot, we should NOT have to discard snapshots.

Reply:
There's always the use-case of cleaning up ansible-managed snapshots at another point of time than creation. Say you want to create them with ansible during some time-critical operation and want to do the cleanup later. Right now it's not possible (without creating yet another snapshot).

In the end I'm OK with both behaviors, as long as it is documented.